integration/systemd/amazon-ecs-agent.service

```diff
@@ -0,0 +1,138 @@
+Environment=ECS_AGENT_VERSION=v1.17.2
+
+ExecStartPre=-/usr/bin/echo "Amazon ECS Agent systemd unit file version 1.0.0"
+
+# Load an updated ECS Agent, if it exists:
+ExecStartPre=-/bin/sh -c 'test -f /var/cache/ecs/desired-image && \
+    docker load --quiet --input=/var/cache/ecs/desired-image \
+    && rm -f $(cat /var/cache/ecs/desired-image) /var/cache/ecs/desired-image'
+
+# If we don't have an ECS Agent, load from disk, if possible, or Docker Hub:
+ExecStartPre=/bin/sh -c "`docker inspect amazon/amazon-ecs-agent &>/dev/null` \
+    || docker load --quiet -input=/var/cache/ecs/ecs-agent.tar \
+    || docker pull amazon/amazon-ecs-agent:${ECS_AGENT_VERSION}"
+
+# Add the IPTable rules needed to enable IAM Roles for Tasks.
+ExecStartPre=/sbin/iptables -t nat -A PREROUTING -d 169.254.170.2/32 \
+    -p tcp -m tcp --dport 80 -j DNAT --to-destination 127.0.0.1:51679
+ExecStartPre=/sbin/iptables -t nat -A OUTPUT -d 169.254.170.2/32 \
+    -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 51679
+
+# Allow the port proxy to route traffic using loopback addresses
+ExecStartPre=/sbin/sysctl --quiet --write net.ipv4.conf.all.route_localnet=1
+
+# Stop the ECS Agent if it was running. Docker stop is used
+# as it will send a SIGTEM and wait 10 seconds before sending SIGKILL
+ExecStartPre=-/bin/sh -c "`docker stop --time 10 ecs-agent &>/dev/null`"
+
+# Remove the container named "ecs-agent"
+ExecStartPre=-/bin/sh -c "`docker rm ecs-agent &>/dev/null`"
+
+# Create the directories needed for the ECS Agent.
+ExecStartPre=-/bin/mkdir -p /var/lib/ecs/dhclient /var/ecs-data
+
+# The ECS Agent is started via docker run
+ExecStart=/usr/bin/docker run --name ecs-agent \
+    --init \
+    --detach=false \
+    --net=host \
+    --pid=host \
+    --cap-add=SYS_ADMIN \
+    --cap-add=NET_ADMIN \
+    --volume=/var/run:/var/run \
+    --volume=/var/log/ecs/:/log \
+    --volume=/var/lib/ecs/data:/data \
+    --volume=/etc/ecs:/etc/ecs \
+    --volume=/sbin:/sbin \
+    --volume=/lib:/lib \
+    --volume=/lib64:/lib64 \
+    --volume=/usr/lib:/usr/lib \
+    --volume=/proc:/host/proc \
+    --volume=/sys/fs/cgroup:/sys/fs/cgroup \
+    --volume=/var/lib/ecs/dhclient:/var/lib/dhclient \
+    --volume=/run/docker/execdriver/native:/var/lib/docker/execdriver/native:ro \
+    --publish=127.0.0.1:51678:51678 \
+    --env ECS_UPDATES_ENABLED=false \
+    --env ECS_DATADIR=/data \
+    --env ECS_ENABLE_TASK_IAM_ROLE=true \
+    --env ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST=true \
+    --env ECS_ENABLE_TASK_ENI=true \
+    --env ECS_LOGFILE=/log/ecs-agent.log \
+    --env ECS_AVAILABLE_LOGGING_DRIVERS=["json-file","syslog","awslogs","none"] \
+    --env ECS_CGROUP_PREFIX=ecs \
+    --log-driver=journald \
+    --env-file=/etc/ecs/ecs.config \
+    amazon/amazon-ecs-agent:${ECS_AGENT_VERSION}
+
+# Docker stop is used as it will send a SIGTEM and wait 10 seconds
+# before sending SIGKILL
+ExecStartPre=-/bin/sh -c "`docker stop --time 10 ecs-agent &>/dev/null`"
```

Comment:
Why do we do this in ExecStartPre? Do we expect the service to already be running when we try to start it? Further, this occurs later in the file than the other ExecStartPre call to 'docker rm'.